anti_patterns/deref.md

```diff
@@ -44,7 +49,7 @@ impl Deref for Bar {
     }
 }
 
-fn main() {
+fn baz() {
```

Comment:
why rename main?

Reply:
I just tried and tests works if the function is named `main'.

Reply:
Might have misunderstood this comment wrong https://github.com/rust-unofficial/patterns/pull/23#discussion_r72037199 when skimming through the older PRs. I thought there shouldn't be `main( )` in examples

Reply:
And llogiq did it ins his initial PR as well:
https://github.com/rust-unofficial/patterns/pull/51/files

So I thought this was kind of a convention or so. Not totally sure though.

Reply:
Just opened a random page of a random rust book. It's full of main functions:
https://rust-lang.github.io/async-book/01_getting_started/04_async_await_primer.html

having the main is good because you can quickly copy paste the code and run it.
If there are two functions it also shows which is the entrypoint of the program.

Reply:
Fixed it where applicable. I'm not sure if we should name functions that take a Mutex or a mutable variable main though? There are some cases still in here.